### ISSUE TITLE
No way to use injector not relying on deprecated `InjectFlags`

### ISSUE THREAD

**Issue by angelaki** [angular/angular#47164]
### Description

If I'm not mistaken the Injector only has these two get methods:

/**
 * Retrieves an instance from the injector based on the provided token.
 * @returns The instance from the injector if defined, otherwise the `notFoundValue`.
 * @throws When the `notFoundValue` is `undefined` or `Injector.THROW_IF_NOT_FOUND`.
 */
abstract get<T>(token: ProviderToken<T>, notFoundValue?: T, flags?: InjectFlags): T;
/**
 * @deprecated from v4.0.0 use ProviderToken<T>
 * @suppress {duplicate}
 */
abstract get(token: any, notFoundValue?: any): any;
/**
 * @deprecated from v5 use the new signature Injector.create(options)
 */

So if I want to set some options I **need** to use the `InjectFlags`. These are marked deprecated but right now only the `inject()` function accepts the InjectOptions:

export declare function inject<T>(token: ProviderToken<T>, options: InjectOptions & {
    optional?: false;
}): T;

Hope putting this in the documentation category is ok, since no code is actually involved so its not really a bug?. Btw.: why is the optional flag added on top of the InjectOptions, already containing this property?

### Is this a browser-specific issue? If so, please specify the device, browser, and version.

```true
Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1401.2
@angular-devkit/build-angular   14.1.2
@angular-devkit/core            14.1.2
@angular-devkit/schematics      14.1.2
@schematics/angular             14.1.2
rxjs                            7.5.6
typescript                      4.7.4
```

**Comment by JoostK** [comment]
The deprecation of `InjectFlags` does indeed appear to be premature, as the `Injector` API does not yet offer an alternative. It is being implemented in #46761, however that's a breaking change that cannot land before v15.

**Comment by angelaki** [comment]
Ah, ok! So you are aware of it already! Thank you so much!
